### ISSUE TITLE
Some stitches dropping images

### ISSUE THREAD

**Issue by JohnDMcMaster** [SiliconAnalysis/xystitch#21]
Ex: https://siliconpr0n.org/map/yamaha/ymf292-f/mz_mit20x2/

Saw this recently after switching image size. See if can figure out where this is coming from

**Comment by JohnDMcMaster** [comment]
Added some better status logging. The net size doesn't match the supertile size which might be a problem related to this

```
2021-04-21T22:48:36.712771: Status w/ 0 / 1 supertiles complete, 0 / 8010 tiles complete
2021-04-21T22:48:36.712816:   0 / 0 submitted tasks complete
2021-04-21T22:48:36.712848:   Net size: 493 MP, ST size: 430 MP
2021-04-21T22:48:36.712884:   mem_net_last 0.000 GB
2021-04-21T22:48:36.712900:   mem_net_max 0.000 GB
2021-04-21T22:48:36.712916:   mem_worker_max 0.000 GB
```

**Comment by JohnDMcMaster** [comment]
Log message was faulty, now it says:

```
2021-04-21T23:15:08.513886: Status w/ 0 / 1 supertiles complete, 0 / 8010 tiles complete
2021-04-21T23:15:08.513905:   0 / 1 submitted tasks complete
2021-04-21T23:15:08.513934:   Net size: 493 MP, ST max size: 835 MP
2021-04-21T23:15:08.513954:   mem_net_last 0.257 GB
2021-04-21T23:15:08.513972:   mem_net_max 0.257 GB
2021-04-21T23:15:08.513988:   mem_worker_max 0.049 GB
```

**Comment by JohnDMcMaster** [comment]
Possibly related to https://github.com/JohnDMcMaster/xystitch/pull/25

Going to close for now, reopen if I see again

**Comment by JohnDMcMaster** [comment]
Looks like this occurred for the image that got *multiple* Dijkstra failures

```
2021-04-25T06:55:37.881589: 2021-04-25T06:55:37.881555 w0: enblend: info: loading next image: /mnt/m10_4/tmp/F050E72D61FD4D1F/st_000300x_013542y_67EC8F3771324B93/0008.tif 1/1
2021-04-25T06:56:55.217635: 2021-04-25T06:56:55.217607 w0: enblend: warning: unable to run Dijkstra optimizer
2021-04-25T06:56:55.217842: 2021-04-25T06:56:55.217830 w0: enblend: note: seam-line end point outside of cost-image
2021-04-25T06:56:55.217993: 2021-04-25T06:56:55.217981 w0: enblend: note: contour #1 of 2, segment #1 of 1, vertex #1550 of 1554
2021-04-25T06:56:55.328016: 2021-04-25T06:56:55.327997 w0: enblend: warning: unable to run Dijkstra optimizer
2021-04-25T06:56:55.328066: 2021-04-25T06:56:55.328056 w0: nblend: note: seam-line end point outside of cost-image
2021-04-25T06:56:55.328098: 2021-04-25T06:56:55.328090 w0: nblend: note: contour #2 of 2, segment #1 of 1, vertex #1575 of 1576
```

Its not immediately obvious to me why it doesn't like this image

**Comment by JohnDMcMaster** [comment]
"If you really want this script to always succeed, then you can turn off seam optimisation with the enblend --nooptimize parameter. Seam optimisation is only useful if you have moving objects in the scene."

Hmm...

**Comment by JohnDMcMaster** [comment]
Just experimented with

```
"enblend": {
                "opts": "--primary-seam-generator=nearest-feature-transform --fine-mask"
```

Which seems to have resolved the issue. I think it did take a lot longer though. Maybe a good time to collect some performance data to compare before making this the default.

**Comment by JohnDMcMaster** [comment]
The issue is pretty well understood now. I've added some notes to the readme on recommendations

TLDR: newer enblend moved to openmp / new algorithm which seems to have a number of side effects. README suggests installing older enblend which still has -m
